pandas/core/frame.py

```diff
@@ -8285,7 +8285,8 @@ def combine(
             result[col] = arr
 
         # convert_objects just in case
-        return self._constructor(result, index=new_index, columns=new_columns)
+        result = self._constructor(result, index=new_index, columns=new_columns)
+        return result.__finalize__(self, method="combine")
```

Comment:
```suggestion
        frame_result = self._constructor(result, index=new_index, columns=new_columns)
        return frame_result.__finalize__(self, method="combine")
```

Will help fix a typing failure

Reply:
Fixed! Thank you.